pkg/controller/decomission.go

```diff
@@ -116,7 +116,7 @@ func (c *Controller) checkForPoolDecommission(ctx context.Context, key string, t
 			return nil, ErrMinIORestarting
 		}
 
-		return nil, nil
```

Comment:
@cniackz this was the bug, we can't be returning err as nil and as well as real value as nil from a function.

When a function returns like this caller doesn't know what to do.

Reply:
Thank you @harshavardhana for letting me know.